### ISSUE TITLE
PCLs can't target 'ASP.NET Core 5.0' anymore.

### ISSUE THREAD

**Issue by tenor** [aspnet/dnx#3237]
I don't know where best to post this issue as there are numerous components involved.

With Visual Studio 2015 Update 1 and Nuget 3.3.0.212, I can no longer create portable class libraries that target 'ASP.NET Core 5.0' and .NET 4.5 anymore.

I'm working on a DNX class library (targeting dnx45 and dnxcore50) that references a PCL library I'm also working on. That PCL library targets .NET 4.5 and 'ASP.NET Core 5.0'.

With VS 2015 Update 1 RC1, I could build nuget packages for the PCL that had a `lib/portable-net45+dnxcore50` folder, which the DNX class library was able to reference fine.
After upgrading to VS2015 Update 1 and rebuilding the nuget package, it now only contains a `lib/portable-net45` folder (no more `+dnxcore50`). The DNX class library now complains that the nuget package doesn't target DNX Core 5.0.

I'm aware of this [announcement](https://github.com/aspnet/Announcements/issues/98); changing the DNX class library's project.json to target dotnet5.4 instead of dnxcore50 doesn't resolve the issue, as there is no `lib\dotnet5.4` folder in the nuget package.

To reproduce this issue, get the latest Json.NET code and run `nuget pack` on the portable solution (which targets ASP.NET Core 5.0'). The produced nuget package will have a `lib\portable-net45+wp80+win8+wpa81` folder. ( no `+dnxcore50`). 
The latest package from the Nuget feed does have a `lib\portable-net45+wp80+win8+wpa81+dnxcore50` folder.

In summary, I'm unable to create a .NET 4.5 PCL nuget package that can be referenced by a DNX class library.
Is there a new way to get this working?

**Comment by Pilchie** [comment]
Plus @rchande and @ericstj

@tenor how are you building the nuget package/nuspec file. 

We changed the registration of ASP.Net Core 5.0, because the old registration was breaking installation of packages into PCLs, but I wouldn't have expected any changes to package generation. 

@yishaigalatzer: does `nuget pack` of a .csproj generate a portable folder based on what the profile is registered as targeting?

**Comment by tenor** [comment]
@Pilchie I dug around some more and found what's causing the problem:

JSON.Net targets profile259. My project targets Profile75.

The \v4.5\Profile\ folder listings for those targets are:
```
[C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETPortable\v4.5\Profile\Profile75\SupportedFrameworks]

    .NET Framework 4.5.xml
    ASP.NET Core 5.0.frameworkxml

[C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETPortable\v4.5\Profile\Profile259\SupportedFrameworks]

    .NET Framework 4.5.xml
    ASP.NET Core 5.0.frameworkxml
    Windows 8.xml
    Windows Phone 8.1.xml
    Windows Phone Silverlight 8.xml
```

If I rename `ASP.NET Core 5.0.frameworkxml` to just have a `.xml` extension and run `nuget pack`, Nuget builds a nuget package with a `lib\portable-net45+dnxcore50+win+wpa81+wp80` folder for JSON.Net (Profile 259).

I'm guessing VS 2015 Update 1 renamed ASP.NET Core 5.0.xml during the upgrade from  Update 1 RC1.

Now, I have new questions:
1. Is there another installer I need, to properly install the ASP.NET Core 5.0 target?
2. Shouldn't the folder name be portable-net45+**netcore50**+win+wpa81+wp80 or portable-net45+**dotnet5.4**+win+wpa81+wp80 or **netstandard1.x**?

**Comment by davidfowl** [comment]
2. You should never build s portable package with a dotnet or net standard moniker m. That's defeats the purpose of it.

**Comment by tenor** [comment]
@davidfowl I'm confused, how about [Announcement/98](https://github.com/aspnet/Announcements/issues/98)?
Aren't we supposed to target **dotnet5.4** in project.json for DNX class libraries?

**Comment by ericstj** [comment]
If you're looking for net45 + dnxcore50 that'd be dotnet5.2 (5.4 is for net46 and later).  You can create a nuspec that constructs the portable- moniker including dnxcore50, but that portable- syntax isn't something that's recommended going forward since it's a closed/fixed set of frameworks (as opposed to dotnet* which is open).

I think your concern here is that nuget.exe pack no longer does the right thing when run on the solution.  I think you'll have to write a nuspec if that is the case.

**Comment by tenor** [comment]
@ericstj I don't think I'm describing the bigger picture properly.
The project I'm working on is [RestBus](https://github.com/tenor/RestBus) and it consists of a bunch of class libraries, as illustrated below:

```
------------------                           ------------------
| RestBus.Others |                           | RestBus.AspNet | -- project.json frameworks: 
------------------                           ------------------    net451, framework5.4
           \                                      /
             \                                   /
               \                                /
                 \      ------------------     /
                   \--> | RestBus.Common | <--/
                        ------------------
                   (portable assembly targeting 
                   .Net 4.5 and ASP.NET Core 5.0)
```

RestBus.Others are other regular (non-portable) .NET 4.5 class libraries in the project that depend on RestBus.Common.
RestBus.AspNet is a DNX class library that depends on RestBus.Common.
RestBus.Common is a .NET 4.5 portable class library that targets ASP.NET Core 5.0

The frameworks in RestBus.AspNet's project.json file are net451 and framework5.4 which I'm interpreting to mean:
I want this DNX class library to be consumable by DNX 451 projects
*and*
I also want this DNX class library to be consumable by DNX Core 5.0 projects.

This setup worked fine until I upgraded to VS 2015 Update 1 (from Update 1 RC), after which new nuget builds of RestBus.Common were rejected by the framework5.4 target of RestBus.AspNet because the +dnxcore50 folder was missing.

I can construct a .nuspec file that includes a +dnxcore50 folder but that feels like I'm forcing DNX to consume the library.

Is it by design that +dnxcore50 is no longer included in Nuget packages of PCLs that target ASP.NET Core 5.0 or is something not properly set up on my computer?

If it's by design, shouldn't DNX applications and class libraries no longer look for the +dnxcore50 folder and walk the dependency graph instead?

**Comment by davidfowl** [comment]
RestBus.Common needs to target a dotnet platform standard (like Eric was saying dotnet5.2) in order to be consumed by a library that specifies another platform standard (dotnet5.4 in this example)

**Comment by tenor** [comment]
@davidfowl That worked! :smile: I added a project.json file that targets dotnet5.2 in RestBus.Common and RestBus.AspNet didn't complain anymore.

Thanks to everyone that looked into the issue.

Should I close this issue? Is the +dnxcore50 folder issue something that needs to be reported? or is it a good thing, since it will lead anyone in the same boat to this thread so they can fix their project the recommended way.

**Comment by PaybackMan** [comment]
This platform is way out of control. Way too much confusion here
